instrumentation-api-caching/src/main/java/io/opentelemetry/instrumentation/api/caching/CacheBuilder.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.caching;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+/** A builder of {@link Cache}. */
+public final class CacheBuilder {
+
+  private final Caffeine<?, ?> caffeine = Caffeine.newBuilder();
+
+  /** Sets the maximum size of the cache. */
+  public CacheBuilder setMaximumSize(long maximumSize) {
+    caffeine.maximumSize(maximumSize);
+    return this;
+  }
+
+  /**
+   * Sets that keys should be referenced weakly. If used, keys will use identity comparison, not
+   * {@link Object#equals(Object)}.
+   */
+  public CacheBuilder setWeakKeys() {
+    caffeine.weakKeys();
+    return this;
+  }
+
+  /** Returns a new {@link Cache} with the settings of this {@link CacheBuilder}. */
+  public <K, V> Cache<K, V> build() {
+    @SuppressWarnings("unchecked")
+    com.github.benmanes.caffeine.cache.Cache<K, V> delegate =
+        (com.github.benmanes.caffeine.cache.Cache<K, V>) caffeine.build();
```

Comment:
Why not add `<K, V>` as type parameters of `CacheBuilder`?

Reply:
Yeah Caffeine has them - since we don't have loading cache at least for now I couldn't find a reason to have them on the builder. This is the same trick Caffeine uses to convert `Caffeine<Object, Object>` to `Cache<K, V>`.